sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -631,8 +631,7 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(
-            AssertNotNull(inputObject, walkedTypePath), fieldName, dataTypeFor(fieldType),
```

Comment:
This change seems ok. Although I'm not totally sure why we add this `AssertNotNull`.

Reply:
Oh, I see. It can be explained by `AssertNotNull`'s doc:

```Note that there are cases where `child.nullable == true`, while we still need to add this assertion...```

That is why the test case failed too...